sceptre/cli/list.py

```diff
@@ -46,12 +46,13 @@ def list_resources(ctx, path):
     write(responses, context.output_format)
 
 
+# flake8: noqa: C901
```

Comment:
@zaro0508  Assuming we agree to leave the rest of the implementation as-is, the question of what to do about the linter remains. The linter is complaining that this function (with my additional branches) is now too complex:

<img width="505" alt="Screenshot 2023-08-31 at 3 29 40 pm" src="https://github.com/Sceptre/sceptre/assets/2305330/e3d43514-38b8-43be-9eb6-a8f4655ce729">

I would tend to argue that the linter is wrong! I think it's quite readable and easy to understand and if I am to refactor, we are going to end up with helper functions somewhere that will be dedicated to this particular use-case and not make much sense.

I could explore the possibility of a nested function perhaps? Or just leave it as-is.

Reply:
i think flake8 is really complaining about the multiple nested for loops in this function.  hmm, i'm a bit torn because i do agree that it's pretty readable but I don't like the idea of disabling the flake8 check for the entire list_resources function.  How about creating a helper function in this file then disabling the check only on the helper function? 

another option might be to use the [python itertools.product](https://docs.python.org/3.8/library/itertools.html#itertools.product) function to reduce the number of loops?

Reply:
Concerning your comment here https://github.com/Sceptre/sceptre/pull/1357#discussion_r1311894136 let me have a look at what the options are. Note that it only seems possible to disable this check for the whole file.

Reply:
@zaro0508 Have a look now. I think it looked better before, but I did find a way to satisfy the linter. (I also added tests, but have not amended the comment, see my other comment.)

Reply:
(Also, I don't think the C901 issue is with nested for loops specifically (and my additional has not actually changed the nesting of for loops), but cyclomatic complexity which I understand to be number of logical pathways through a function. I investigated the itertools function but I don't think this is a use-case for that?)